src/utils/filterTypes.ts

```diff
@@ -0,0 +1,51 @@
+export type Filter = {
+  campo: string;
+  operador: string;
+  valor: string | number | string[];
+};
+
+export type Sorter = {
+  campo: string;
+  ordem: 'ASC' | 'DESC';
+};
+
+export function buildWhereClause(filters: Filter[]) {
+  const where: Record<string, unknown> = {};
+
+  if (!filters || filters.length === 0) {
+    return where;
+  }
+
+  filters.forEach(filter => {
+    const { campo, operador, valor } = filter;
+
+    switch (operador) {
+      case 'like':
+        where[campo] = { contains: valor, mode: 'insensitive' };
+        break;
+      case '>=':
+        where[campo] = { gte: valor };
+        break;
+      case '<=':
+        where[campo] = { lte: valor };
+        break;
+      case 'in':
+        where[campo] = { in: valor };
+        break;
+      default:
+        where[campo] = valor;
+    }
+  });
+
+  return where;
+}
+
+export function buildOrderClause(sorters: Sorter[]) {
+  if (!sorters || sorters.length === 0) {
+    return undefined;
+  }
+
```

Comment:
aqui a mesma coisa do comentário acima, testar se funciona para os casos em que não temos filters ou sorters
```suggestion
export function buildOrderClause(sorters: Sorter[] = []) {
```

Reply:
Tudo ok.